### PR TITLE
Chemtab Dynamic Source Prediction

### PR DESCRIPTION
**Pull request by chandola**
Updated ChemTab model that can output source terms along with source energy

### CODE REVIEW

File: src/chemistry/chemTabModel.cpp [hunk 1]
```diff
@@ -62,6 +59,22 @@ ablate::chemistry::ChemTabModel::ChemTabModel(std::filesystem::path path) {
     inputFileStream.open(ipath.c_str(), std::ios::in);
     LoadBasisVectors(inputFileStream, speciesNames.size(), iWmat);
     inputFileStream.close();
+    //load source energy scaler
+    sourceEnergyScaler = (PetscReal *) malloc(2*sizeof(PetscReal));
```
Comment (mmcgurn) [inline src/chemistry/chemTabModel.cpp:63]: I think that we missed freeing the sourceEnergyScaler

File: src/chemistry/chemTabModel.cpp [hunk 2]
```diff
@@ -62,6 +59,22 @@ ablate::chemistry::ChemTabModel::ChemTabModel(std::filesystem::path path) {
...
+
+    std::getline(inputFileStream, line);
+    std::stringstream lineStream2(line);
+    getline(lineStream2,value,' ');
+    sourceEnergyScaler[1] = std::stod(value);
+
+    inputFileStream.close();
 }
 
 ablate::chemistry::ChemTabModel::~ChemTabModel() {
```
Comment (mmcgurn) [inline src/chemistry/chemTabModel.cpp:80]: Do we need to iterate through each component of Wmat and iWmat and free each component?

**Review by mmcgurn — CHANGES_REQUESTED** [review]
Thanks for submitting the PR!.  I added a question about memory malloc/free if you don't mind taking a look.  You will need to rebase this branch based upon the latest main and bump the version (https://ablate.dev/content/development/UsingGitWithABLATE.html#contributing-code-aka-pull-request).  I think that you will need to run the code through clang format to get the formatting to pass the tests.  Please let me know if you have any questions or run into any problems.

File: src/chemistry/chemTabModel.cpp (hunk 2 above)
Comment (chandola) [reply src/chemistry/chemTabModel.cpp:80]: Yes, I will add that.

File: src/chemistry/chemTabModel.cpp (hunk 1 above)
Comment (chandola) [reply src/chemistry/chemTabModel.cpp:63]: Will free the variable.

**Review by mmcgurn — APPROVED** [review]

**Comment by chandola** [comment]
I have fixed the version number in the new commit.

**Pull request merged**